app/concepts/todo/create.rb

```diff
@@ -10,41 +9,82 @@ class Todo::Create < Trailblazer::Operation
   #         create a default `todo_list`
   #         associate `todo_list` with `user`
   #         associate `todo` with `todo_list`
+  # def process(params)
+  #   # make sure title is not blank
+  #   return invalid! if params[:todo][:title].blank?
+  #
+  #   # build the todo
+  #   todo = @model = Todo.new(params.require(:todo).permit(:title, :description))
+  #
+  #   # find or create the parent list for the todo
+  #   if params[:todo_list_id].blank?
+  #     todo_list = TodoList.find_or_create_by(name: "Default To-do List")
+  #
+  #     # Find or create the owner of the todo_list:
+  #     ##### If current_user_id is present, fetch that user
+  #     ##### otherwise, create a new one
+  #     current_user =
+  #       if params[:current_user] == nil
+  #         if params[:current_user_id] == nil
+  #           User.create!(fullname: "Guest")
+  #         else
+  #           User.find(params[:current_user_id])
+  #         end
+  #       else
+  #         params[:current_user]
+  #       end
+  #
+  #     # assign the todo_list to the user
+  #     todo_list.user = current_user
+  #     todo_list.save!
+  #   else # i.e. if params[:todo_list_id] is not nil
+  #     todo_list = TodoList.find(params[:todo_list_id])
+  #   end
+  #
+  #   # associate the todo with the todo_list
+  #   todo.list = todo_list
+  #   todo.save!
+  # end
+
+  contract do
+    property :title, validates: {presence: true, length: {maximum: 160}}
+  end
+
   def process(params)
-    # make sure title is not blank
-    return invalid! if params[:todo][:title].blank?
-
-    # build the todo
-    todo = @model = Todo.new(params.require(:todo).permit(:title, :description))
-
-    # find or create the parent list for the todo
-    if params[:todo_list_id].blank?
-      todo_list = TodoList.find_or_create_by(name: "Default To-do List")
-
-      # Find or create the owner of the todo_list:
-      ##### If current_user_id is present, fetch that user
-      ##### otherwise, create a new one
-      current_user =
-        if params[:current_user] == nil
-          if params[:current_user_id] == nil
-            User.create!(fullname: "Guest")
-          else
-            User.find(params[:current_user_id])
-          end
+    # There's an issue on strong parameters with rspec
+    # Comment this line below for rspec
+    todo_params = params.require(:todo).permit(:title, :description)
+    # Uncomment this line below for rspec
+    # todo_params = params[:todo]
+
+    todo = Todo.create(todo_params)
+
+    validate(todo_params, todo) do
```

Comment:
This is correct and there's a shortcut for doing this. Can you search about `Trailblazer::Operation::Model`